### PR TITLE
Fix empty project displayed in issue sidebar

### PR DESCRIPTION
**Pull request by yp05327**
You can confirm this issue in https://try.gitea.io/yp05327/testrepo/issues/2

Before:
image
After:
image

This issue comes from the change in #25468.
`LoadProject` will always return at least one record, so we use `ProjectID` to check whether an issue is linked to a project in the old code.
As other `issue.LoadXXX` functions, we need to check the return value from `xorm.Session.Get`.

In recent unit tests, we only test `issueList.LoadAttributes()` but don't test `issue.LoadAttributes()`. So I added a new test for `issue.LoadAttributes()` in this PR.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: models/issues/issue_project.go [hunk 1]
```diff
@@ -16,13 +16,14 @@ import (
...
 		var p project_model.Project
-		if _, err = db.GetEngine(ctx).Table("project").
+		has, err := db.GetEngine(ctx).Table("project").
 			Join("INNER", "project_issue", "project.id=project_issue.project_id").
-			Where("project_issue.issue_id = ?", issue.ID).
-			Get(&p); err != nil {
+			Where("project_issue.issue_id = ?", issue.ID).Get(&p)
+		if err != nil {
 			return err
+		} else if has {
```
Comment (puni9869) [inline models/issues/issue_project.go:24]:
```suggestion
		} 
		if has {
```
Not a blocker. But these are two different checks

**Review by puni9869 — APPROVED** [review]

File: models/issues/issue_project.go (hunk 1 above)
Comment (yp05327) [reply models/issues/issue_project.go:24]: They look the same to me. 🤔

Comment (puni9869) [reply models/issues/issue_project.go:24]: Technically you are 💯  correct.

From the readability point of view`if ... else if ... else` should be act on same variable.
Here `if` block is related to `err` so `else if` also point to `err`
and for `has` it should be a different `if` block.

PS: No blocker my side. Just a thought. :)

**Comment by yp05327** [comment]
Wait a minite. There's a problem.
In some places, we don't check whether issue.Project is nil, before use it.
image

**Comment by yp05327** [comment]
Issue(ID:2) was removed here:
https://github.com/go-gitea/gitea/pull/10814/files#diff-8f8d49e4029a3d68849f2af16dd68a6dcbd2bb25d9ba6d6b906185c1a8cb0f3cL37
But we have a test for it 😕 
image

**Review by puni9869 — APPROVED** [review]

**Comment by yp05327** [comment]
Added a new unit test for `issue.LoadAttributes()`, it will fail in old code:
image

File: models/issues/issue_project.go [hunk 2]
```diff
@@ -16,13 +16,14 @@ import (
 func (issue *Issue) LoadProject(ctx context.Context) (err error) {
 	if issue.Project == nil {
 		var p project_model.Project
-		if _, err = db.GetEngine(ctx).Table("project").
+		has, err := db.GetEngine(ctx).Table("project").
```
Comment (denyskon) [inline models/issues/issue_project.go:19]: I think `has` as variable name is not ideal, what about something like `res` (for result/response)?

File: routers/web/org/projects.go [hunk 3]
```diff
@@ -436,9 +436,11 @@ func UpdateIssueProject(ctx *context.Context) {
 
 	projectID := ctx.FormInt64("id")
 	for _, issue := range issues {
-		oldProjectID := issue.Project.ID
-		if oldProjectID == projectID {
-			continue
+		if issue.Project != nil {
+			oldProjectID := issue.Project.ID
+			if oldProjectID == projectID {
```
Comment (denyskon) [inline routers/web/org/projects.go]: What's the point of assigning a new variable just for the if check?
Why not just do
```suggestion
			if issue.Project.ID == projectID {
```

File: routers/web/repo/projects.go [hunk 4]
```diff
@@ -385,9 +385,11 @@ func UpdateIssueProject(ctx *context.Context) {
 
 	projectID := ctx.FormInt64("id")
 	for _, issue := range issues {
-		oldProjectID := issue.Project.ID
-		if oldProjectID == projectID {
-			continue
+		if issue.Project != nil {
+			oldProjectID := issue.Project.ID
+			if oldProjectID == projectID {
```
Comment (denyskon) [inline routers/web/repo/projects.go]: As above, why not just do
```suggestion
			if issue.Project.ID == projectID {
```

File: models/issues/issue_project.go (hunk 2 above)
Comment (delvh) [reply models/issues/issue_project.go:19]: 1. `has` is almost idiomatic within the gitea code
2. `res` is not what's returned here: The function returns `bool, err`, not `<any>, err` to show whether the pointer got a new value (see 3.)
3. The actual result is set within the function as a pointer is explicitly passed: The `&p` means that the variable `p` is changed directly.

Comment (delvh) [reply models/issues/issue_project.go:19]: The only potential alternative to `has` that would make sense and is also used throughout the code is `exists` which is probably a better name for it.

Comment (denyskon) [reply models/issues/issue_project.go:19]: Oh okay, didn't know that it is used in Gitea 😆

File: routers/web/org/projects.go (hunk 3 above)
Comment (yp05327) [reply routers/web/org/projects.go]: I didn't notice this. Just added Line 439 here. 😄

But,
in old code, `oldProjectID` comes from `issue.ProjectID()` which is a function.
Maybe we don't need this any more.

**Review by denyskon — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**